app/services/notification_service.rb

```diff
@@ -0,0 +1,76 @@
+class NotificationService
+  attr_accessor :user, :matching_users
+
+  def initialize(user, matching_users)
+    @user = user
+    @matching_users = matching_users
+
+    set_id unless user.nil?
```

Comment:
Hmm not sure if updating the user here is a good approach 🤔

Reply:
I do agree with this. 
Just followed @jiv upper recommendation, I might have understood it all wrong tho. Thoughts? @gonzaM @jiv

Reply:
we talked about this in person. I suggested adding the clause only in the notify, not here. Good call @gonzaM 👍